packages/sync-service/lib/electric/shapes/api.ex

```diff
@@ -319,24 +319,15 @@ defmodule Electric.Shapes.Api do
   end
 
   defp hold_until_stack_ready(%Api{} = api) do
-    stack_id = stack_id(api)
-
-    ref =
-      Electric.StackSupervisor.subscribe_to_stack_events(
-        api.stack_events_registry,
-        stack_id
-      )
+    api
+    |> stack_id()
+    |> Electric.StatusMonitor.wait_until_active(api.stack_ready_timeout)
+    |> case do
+      :ok ->
+        :ok
 
-    if Electric.ProcessRegistry.alive?(stack_id, Electric.Replication.Supervisor) do
-      :ok
-    else
-      receive do
-        {:stack_status, ^ref, :ready} ->
-          :ok
-      after
-        api.stack_ready_timeout ->
-          {:error, Response.error(api, "Stack not ready", status: 503)}
-      end
+      {:error, :timeout} ->
```

Comment:
Maybe add a logger warning here with a concrete number?

```suggestion
      {:error, :timeout} ->
        Logger.warning("Stack not ready after #{api.stack_ready_timeout}ms")
```